cypress/component/QueryForm.cy.tsx

```diff
@@ -155,54 +156,64 @@ describe('QueryForm', () => {
   });
   it('Fires updateContinuousQueryParams event handler with the appropriate payload when a continuous field is selected', () => {
     const updateContinuousQueryParamsSpy = cy.spy().as('updateContinuousQueryParamsSpy');
-    cy.mount(
-      <QueryForm
-        availableNodes={props.availableNodes}
-        diagnosisOptions={props.diagnosisOptions}
-        assessmentOptions={props.assessmentOptions}
-        selectedNode={props.selectedNode}
-        minAge={props.minAge}
-        maxAge={props.maxAge}
-        sex={props.sex}
-        diagnosis={props.diagnosis}
-        minNumImagingSessions={props.minNumImagingSessions}
-        minNumPhenotypicSessions={props.minNumPhenotypicSessions}
-        assessmentTool={props.assessmentTool}
-        imagingModality={props.imagingModality}
-        pipelineVersion={props.pipelineVersion}
-        pipelineName={props.pipelineName}
-        pipelines={props.pipelines}
-        updateCategoricalQueryParams={props.updateCategoricalQueryParams}
-        updateContinuousQueryParams={updateContinuousQueryParamsSpy}
-        loading={props.loading}
-        onSubmitQuery={props.onSubmitQuery}
-      />
-    );
+    // Provide local state so the controlled form reflects typing; without feeding the updated
+    // value back in, the field would instantly revert to the default string and the change handler would
+    // only ever report intermediate characters.
+    function QueryFormWithState(): JSX.Element {
+      const [minAge, setMinAge] = useState<string>(defaultProps.minAge);
+      return (
+        <QueryForm
+          availableNodes={defaultProps.availableNodes}
+          diagnosisOptions={defaultProps.diagnosisOptions}
+          assessmentOptions={defaultProps.assessmentOptions}
+          selectedNode={defaultProps.selectedNode}
+          minAge={minAge}
+          maxAge={defaultProps.maxAge}
+          sex={defaultProps.sex}
+          diagnosis={defaultProps.diagnosis}
+          minNumImagingSessions={defaultProps.minNumImagingSessions}
+          minNumPhenotypicSessions={defaultProps.minNumPhenotypicSessions}
+          assessmentTool={defaultProps.assessmentTool}
+          imagingModality={defaultProps.imagingModality}
+          pipelineVersion={defaultProps.pipelineVersion}
+          pipelineName={defaultProps.pipelineName}
+          pipelines={defaultProps.pipelines}
+          updateCategoricalQueryParams={defaultProps.updateCategoricalQueryParams}
+          updateContinuousQueryParams={(label, value) => {
+            setMinAge(value);
+            updateContinuousQueryParamsSpy(label, value);
+          }}
+          loading={defaultProps.loading}
+          onSubmitQuery={defaultProps.onSubmitQuery}
```

Comment:
Same comment here as for the continuousField test: If the component doesn't have state, then keeping state is not its concern anymore and we don't need to test that / simulate it. If the thing you're really concerned over is whether the component can handle the second (or third) character someone enters, then just initialize that state by setting the value explicitly.